src/menu/DatasetSearchBox.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import {Button, Col, Row} from 'react-bootstrap';
+import {RestBaseUtils, RestStorageUtils, StoreUtils} from 'tdp_core';
+import {Species, SpeciesUtils} from 'tdp_gene';
+import {FormatOptionLabelMeta} from 'react-select';
+import {AsyncPaginate} from 'react-select-async-paginate';
+import Highlighter from 'react-highlight-words';
+import {I18nextManager, IDTypeManager, UserSession} from 'phovea_core';
+import {IDataSourceConfig} from 'tdp_publicdb';
+import {GraphContext, SESSION_KEY_NEW_ENTRY_POINT} from 'ordino';
+
+interface IDatasetSearchBoxProps {
+    placeholder: string;
+    viewId: string;
+    datasource: IDataSourceConfig;
+}
+
+export function DatasetSearchBox({placeholder, datasource, viewId}: IDatasetSearchBoxProps) {
+    const {db, base, dbViewSuffix, entityName, idType: idtype} = datasource;
+    const [items, setItems] = React.useState<IdTextPair[]>(null);
+    const {manager} = React.useContext(GraphContext);
+    const search = (query: string): Promise<{more: boolean, items: Readonly<IdTextPair>[]}> => {
+        return RestBaseUtils.getTDPLookup(db, base + dbViewSuffix, {
+            column: entityName,
+            species: SpeciesUtils.getSelectedSpecies(),
+            query
+        });
+    };
+
+
+    const loadOptions = async (inputValue: string, _, {page}: {page: number}) => {
+        const options = await search(inputValue);
+        return {
+            options: options.items,
+            hasMore: options.more,
+            additional: {
+                page: page + 1,
+            },
+        };
+    };
+
+    function formatOptionLabel(option: IdTextPair, ctx: FormatOptionLabelMeta<IdTextPair, true>) {
+        if (!ctx.inputValue || ctx.selectValue.some((o) => o.id === option.id)) {
+            return option.text;
+        }
+        return (
+            <Highlighter
+                highlightClassName="YourHighlightClass"
```

Comment:
This is not used at the moment since we do not apply a custom style to the highlighted element. We use the default style similar to the `<mark>` effect, so we can remove it entirely.